### PR TITLE
Azure Alerts

### PR DESCRIPTION
**Pull request by eedorenko**
ARM templates with the alerts defined in [ADR](https://github.com/Azure/iotedge-lorawan-starterkit/blob/docs/main/docs/adr/005_observability.md):
 - HighUpstreamMessageLatency
 - HighErrorCount
 - HighReceiveWindowMisses
 - HighDownstreamMessagesLostRatio

Issue #891

### CODE REVIEW

File: Template/azureMonitorAlerts.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                    "allOf": [
+                        {
+                            "alertSensitivity": "[variables('alertSensitivity')]",
+                            "failingPeriods": {
+                                "numberOfEvaluationPeriods": "[variables('evaluationPeriods')]",
+                                "minFailingPeriodsToAlert": "[variables('failingPeriods')]"
+                            },
+                            "name": "Metric1",
+                            "metricNamespace": "Microsoft.Devices/IotHubs",
+                            "metricName": "c2d.commands.egress.abandon.success",
```
Comment (bastbu) [inline Template/azureMonitorAlerts.json:179]: Even though it's not in the ADR, should we maybe consider rejected messages here as well?

**Review by bastbu — APPROVED** [review]

File: Template/azuredeploy.json [hunk 2]
```diff
@@ -269,6 +269,30 @@
             },
             "dependsOn": []
         },
+        {
+            "name": "azuremonitoralerts",
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2015-01-01",
```
[the remark this answers is not in the document]
Comment (eedorenko) [reply Template/azuredeploy.json:275]: It's also autogenerated when you're exporting the template in the portal.

File: Template/azureMonitorAlerts.json (hunk 1 above)
Comment (eedorenko) [reply Template/azureMonitorAlerts.json:179]: That's a good point. I'll add a new alert.

File: Template/azureMonitorAlerts.json [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                "windowSize": "[variables('aggregationPeriod')]",
+                "criteria": {
+                    "allOf": [
+                        {
+                            "alertSensitivity": "[variables('alertSensitivity')]",
+                            "failingPeriods": {
+                                "numberOfEvaluationPeriods": "[variables('evaluationPeriods')]",
+                                "minFailingPeriodsToAlert": "[variables('failingPeriods')]"
+                            },
+                            "name": "Metric1",
```
[the remark this answers is not in the document]
Comment (eedorenko) [reply Template/azureMonitorAlerts.json:92]: It's autogenerated when you're exporting the template in the portal. When an alert has multiple conditions based on multiple metrics it gives names to those conditions like Metric1,Metric2, ...
Screen Shot 2021-12-06 at 10 08 20 AM

Don't think we should/need change them.

File: Template/azureMonitorAlerts.json (hunk 1 above)
Comment (eedorenko) [reply Template/azureMonitorAlerts.json:179]: Added an alert based on c2d.commands.egress.reject.success

File: Template/azuredeploy.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (eedorenko) [reply Template/azuredeploy.json]: Updated to 2021-04-01 and added a variable

**Comment by eedorenko** [comment]
Added "High Upstream Messages Lost Ratio" on top of D2CMessagesLostRatio metric

**Pull request merged**